source/common/network/socket_impl.h

```diff
@@ -7,6 +7,49 @@
 namespace Envoy {
 namespace Network {
 
+namespace SocketInterface {
```

Comment:
Do you want this to be an actual abstract class that you can inject later? Or you want to do that as a follow up? Ultimately I suspect you want this to hang off of the `Api` interface where we can handle platform injection more easily. Not sure if it's better to do it now or wait until later. I don't feel strongly about it.

Reply:
I was planning on leaving it for later, especially because I'm not entirely sure about the best approach. 

Originally, I was hoping I could hide all of the "raw" socket calls behind the SocketImpl constructors, but then I stumbled upon some places that only care about the IoHandles (and rely on syscalls to retrieve addresses and whatnot). So, I decided to postpone the decision until I completely migrate everything to SocketImpl.

Reply:
OK we can do the injection later.